src/flatbuffer_types.rs

```diff
@@ -26,7 +26,8 @@ use crate::flatbuffer_types::manifest_generated::{
 };
 use crate::manifest::{Manifest, ManifestCodec};
 
-pub(crate) struct SsTableIndexOwned {
+/// A wrapper around a `Bytes` buffer containing a FlatBuffer-encoded `SsTableIndex`.
+pub struct SsTableIndexOwned {
```

Comment:
would you mind sharing your perspective on changing this struct from `pub(crate)` to `pub`?

Reply:
I got a warning because it's referenced by a public struct, so it should be public too: https://github.com/slatedb/slatedb/pull/231/files#diff-3fe6409ce28a780ab93ab98688c8cd28908dce9e7a60c5caa6d20e8910d15d64R13

But if this should be private, maybe we need some kind of intermediary object to expose publicly.

Reply:
do you have any recommendation to avoid this?

Reply:
I've created wrappers to prevent leaking internal structs.